libs/clients/license-client/src/lib/clients/machine-license-client/machineLicenseMapper.ts

```diff
@@ -4,42 +4,30 @@ import {
 } from '@island.is/clients/adr-and-machine-license'
 
 import format from 'date-fns/format'
-import isAfter from 'date-fns/isAfter'
 import { Locale } from '@island.is/shared/types'
 import is from 'date-fns/locale/is'
 import enGB from 'date-fns/locale/en-GB'
 import { format as formatSsn } from 'kennitala'
 
-export const checkLicenseExpirationDate = (license: VinnuvelaDto) => {
-  return license.vinnuvelaRettindi
-    ? license.vinnuvelaRettindi
-        .filter((field) => field.kenna || field.stjorna)
-        .every(
-          (field: VinnuvelaRettindiDto) =>
-            field.kenna &&
-            !isAfter(new Date(field.kenna), new Date()) &&
-            field.stjorna &&
-            !isAfter(new Date(field.stjorna), new Date()),
-        )
-    : null
-}
+const compareDates = (newDate: Date, latestDate?: Date) =>
+  !latestDate || newDate > latestDate ? newDate : latestDate
 
 export const findLatestExpirationDate = (license: VinnuvelaDto) => {
   if (!license.vinnuvelaRettindi) {
     return null
   }
 
-  let maxDate = new Date()
+  let latestDate: Date | undefined
   for (const right of license.vinnuvelaRettindi) {
-    if (right.stjorna && new Date(right.stjorna) > maxDate) {
-      maxDate = new Date(right.stjorna)
+    if (right.stjorna) {
+      latestDate = compareDates(new Date(right.stjorna), latestDate)
```

Comment:
So if I'm understanding correctly each machine license can have multiple different "licenses" that each can have two different expiration date (one for stjorna, one for kenna) and here you are just finding the latest date of all these possible expiration dates? 🤓 And the latest of them all should be the one that decides if the license is expired or not? (Is that how `payload.metadata.expired` is calculated?)